sycl/test-e2e/Matrix/Inputs/common.hpp

```diff
@@ -168,7 +173,18 @@ template <typename T1, typename T2, bool exact = false>
 bool matrix_compare(unsigned int rows, unsigned int cols, T1 *src, T2 *ref) {
   for (int i = 0; i < rows; i++) {
     for (int j = 0; j < cols; j++) {
-      if constexpr (!exact && (std::is_same_v<T1, float> ||
+      if constexpr (!exact && (std::is_same_v<T1, bfloat16> &&
```

Comment:
if `FLOAT_EPSILON` is used instead of `BFLOAT16_EPSILON`, no need to update `matrix_compare`.